cmd/list.go

```diff
@@ -0,0 +1,87 @@
+package cmd
+
+import (
+	"fmt"
+	"github.com/spf13/cobra"
+	"golang.org/x/mod/modfile"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+func ListCmd() *cobra.Command {
+	var listCmd = &cobra.Command{
+		Use:   "list",
+		Short: "List the docker images",
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdFlags := cmd.Flags()
+
+			configFile, _ := cmdFlags.GetString(flagFile)
+			if configFile == "" {
+				// try to load a local chains.yaml, but do not panic for any error, will fall back to embedded chains.
+				cwd, err := os.Getwd()
+				if err == nil {
+					chainsYamlSearchPath := filepath.Join(cwd, "chains.yaml")
+					if err := loadChainsYaml(chainsYamlSearchPath); err != nil {
+						fmt.Printf("No config found at %s, using embedded chains. pass -f to configure chains.yaml path.\n", chainsYamlSearchPath)
+					} else {
+						fmt.Printf("Loaded chains from %s\n", chainsYamlSearchPath)
+					}
+				}
+			} else {
+				// if flag is explicitly provided, panic on error since intent was to override embedded chains.
+				if err := loadChainsYaml(configFile); err != nil {
+					panic(err)
+				}
+			}
+			list()
+		},
+	}
+
+	listCmd.PersistentFlags().StringP(flagFile, "f", "", "chains.yaml config file path (searches for chains.yaml in current directory by default)")
+
+	return listCmd
+}
+
+func list() {
+	requires := []string{
+		"cosmos-sdk",
+		"ibc-go",
+	}
+	for _, chain := range chains {
+		url := fmt.Sprintf("https://raw.githubusercontent.com/%s/%s/main/go.mod",
+			chain.GithubOrganization, chain.GithubRepo)
+		resp, err := http.Get(url)
+		if err != nil {
+			continue
+		}
+		defer resp.Body.Close()
+		body, err := io.ReadAll(resp.Body)
+		if err != nil {
+			continue
+		}
+		mod, err := modfile.Parse("", body, nil)
+		if err != nil {
+			continue
+		}
```

Comment:
If encountering an error, can we print the error so the user will know something went wrong?

Reply:
Here's the full output with errors:

```
Loaded chains from /Users/giuseppe/Developer/heighliner/chains.yaml

acrechain:
  cosmos-sdk@v0.45.10
  ibc-go@v3.2.0

agoric:
  error: not enough repo info

akash:
  error: 404 Not Found

assetmantle:
  error: 404 Not Found

axelar:
  cosmos-sdk@v0.45.11
  ibc-go@v4.2.0

tofnd:
  error: 404 Not Found

basilisk:
  error: 404 Not Found

bitcanna:
  cosmos-sdk@v0.45.15
  ibc-go@v3.4.0

bitsong:
  cosmos-sdk@v0.45.11
  ibc-go@v3.3.1

bostrom:
  cosmos-sdk@v0.45.5
  ibc-go@v3.0.0

burnt:
  cosmos-sdk@v0.45.11
  ibc-go@v4.2.0

carbon:
  error: 404 Not Found

cerberus:
  cosmos-sdk@v0.45.10
  ibc-go@v3.3.0

cheqd:
  cosmos-sdk@v0.46.10
  ibc-go@v6.1.0

chihuahua:
  cosmos-sdk@v0.45.12
  ibc-go@v4.2.0

comdex:
  cosmos-sdk@v0.46.7
  ibc-go@v4.3.0

composable:
  error: 404 Not Found

gaia:
  cosmos-sdk@v0.45.13-ics
  ibc-go@v4.2.0

ics:
  cosmos-sdk@v0.45.11
  ibc-go@v4.2.0

crescent:
  cosmos-sdk@v0.45.10
  ibc-go@v3.4.0

cronos:
  cosmos-sdk@v0.46.11
  ibc-go@v5.2.0

cryptoorgchain:
  error: 404 Not Found

decentr:
  error: 404 Not Found

desmos:
  error: 404 Not Found

dig:
  error: 404 Not Found

duality:
  error: 404 Not Found

emoney:
  error: 404 Not Found

evmos:
  cosmos-sdk@v0.46.11
  ibc-go@v6.1.0

fetchhub:
  error: 404 Not Found

firmachain:
  error: 404 Not Found

gravitybridge:
  error: 404 Not Found

ibc-go-simd:
  cosmos-sdk@v0.47.0

impacthub:
  cosmos-sdk@v0.45.12
  ibc-go@v4.3.0

injective:
  error: not enough repo info

icad:
  error: 404 Not Found

icqd:
  cosmos-sdk@v0.45.8
  ibc-go@v3.0.1

irisnet:
  error: 404 Not Found

juno:
  cosmos-sdk@v0.45.15
  ibc-go@v4.3.0

kichain:
  cosmos-sdk@v0.42.10

konstellation:
  error: 404 Not Found

kujira:
  error: 404 Not Found

likecoin:
  error: 404 Not Found

lumnetwork:
  error: 404 Not Found

nomic:
  error: 404 Not Found

neutron:
  cosmos-sdk@v0.45.11
  ibc-go@v4.2.0

nibiru:
  error: 404 Not Found

noble:
  cosmos-sdk@v0.45.13
  ibc-go@v3.4.0

nolus:
  cosmos-sdk@v0.45.13-ics
  ibc-go@v4.2.0

omniflixhub:
  cosmos-sdk@v0.45.11
  ibc-go@v3.4.0

onomy:
  cosmos-sdk@v0.44.5
  ibc-go@v2.0.4

onomy-gbt:
  error: 404 Not Found

osmosis:
  cosmos-sdk@v0.47.1
  ibc-go@v4.3.0

panacea:
  cosmos-sdk@v0.45.12
  ibc-go@v4.3.0

penumbra:
  error: 404 Not Found

persistence:
  cosmos-sdk@v0.45.14
  ibc-go@v4.3.0

polkadot:
  error: 404 Not Found

provenance:
  cosmos-sdk@v0.46.7
  ibc-go@v6.1.0

quasar:
  cosmos-sdk@v0.45.14
  ibc-go@v4.3.0

quasartestnet:
  error: not enough repo info

quicksilver:
  cosmos-sdk@v0.46.10
  ibc-go@v5.2.0

regen:
  cosmos-sdk@v0.46.7
  ibc-go@v5.2.0
  cosmos-sdk@v0.1.0
  cosmos-sdk@v1.0.0-beta.5
  cosmos-sdk@v1.0.0-alpha.12

rizon:
  error: 404 Not Found

secretnetwork:
  error: not enough repo info

sei:
  error: 404 Not Found

sentinel:
  error: 404 Not Found

shentu:
  error: 404 Not Found

sifchain:
  error: 404 Not Found

sim:
  cosmos-sdk@v1.0.0-beta.1.0.20220726092710-f848e4300a8a

sommelier:
  cosmos-sdk@v0.44.5-patch
  ibc-go@v2.0.0

stargaze:
  cosmos-sdk@v0.45.14
  ibc-go@v4.3.0

starname:
  cosmos-sdk@v0.45.9
  ibc-go@v3.3.0
  cosmos-sdk@v0.4.0-rc

stride:
  cosmos-sdk@v0.46.7
  ibc-go@v5.1.0

tendermint:
  error: no versions found

terra:
  cosmos-sdk@v0.45.12
  ibc-go@v3.4.0

thorchain:
  error: 404 Not Found

umee:
  cosmos-sdk@v0.46.11
  ibc-go@v5.2.0

vidulum:
  cosmos-sdk@v0.45.9
  ibc-go@v1.2.2

wasm:
  cosmos-sdk@v0.47.0
  ibc-go@v7.0.0
```